Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.DynamicExpression.cs

```diff
@@ -49,34 +51,39 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public DynamicExpression Update (IEnumerable<Expression> arguments)
     {
```

Comment:
:thought_balloon: Just because the implementation doesn't crash when `arguments` is null doesn't mean the contract needs to allow null.